packages/react/src/components/SlashMenu/SlashMenuPositioner.tsx

```diff
@@ -33,64 +32,57 @@ export const SlashMenuPositioner = <
 
   const referencePos = useRef<DOMRect>();
 
+  const { refs, update, context, floatingStyles } = useFloating({
+    open: show,
+    placement: "bottom-start",
+    middleware: [flip()],
+  });
+
+  const { isMounted, styles } = useTransitionStyles(context);
+
   useEffect(() => {
     return props.editor.slashMenu.onUpdate((slashMenuState) => {
       setShow(slashMenuState.show);
       setFilteredItems(slashMenuState.filteredItems);
       setKeyboardHoveredItemIndex(slashMenuState.keyboardHoveredItemIndex);
 
       referencePos.current = slashMenuState.referencePos;
+      update();
     });
-  }, [props.editor]);
+  }, [props.editor, update]);
 
-  const getReferenceClientRect = useMemo(
-    () => {
-      if (!referencePos.current) {
-        return undefined;
-      }
-
-      return () => referencePos.current!;
-    },
-    [referencePos.current] // eslint-disable-line
-  );
+  // const { ref, updateMaxHeight } = usePreventMenuOverflow(); // TODO, needed?
```

Comment:
This is indeed still needed - it's basically a fallback for handling overflow. Normally, if there's no space to render the menu in the default `"bottom"` position, it will get flipped to the top. But if the viewport itself is too small to see the whole menu, this hook constrains the height of the menu to fit in the viewport.